server/src/service.rs

```diff
@@ -293,4 +296,57 @@ impl Service {
 
         Ok(())
     }
+
+    pub async fn set_locked(
+        &self,
+        locked: bool,
+        objects: &[OwnedObjectPath],
+    ) -> Result<(Vec<OwnedObjectPath>, Vec<OwnedObjectPath>), ServiceError> {
+        let mut locked_changed = Vec::new();
+        let mut locked_unchanged = Vec::new();
+        let collections = self.collections.lock().await;
+
+        for object in objects {
+            for collection in collections.iter() {
+                let collection_locked = collection.is_locked().await;
+                if object == collection.path() {
+                    if collection_locked == locked {
+                        tracing::debug!(
+                            "Collection: {} is already {}.",
+                            object,
+                            if locked { "locked" } else { "unlocked" }
+                        );
+                        locked_changed.push(object.clone());
```

Comment:
Shouldn't be unchanged? And the same elsewhere.

Reply:
[No](https://github.com/bilelmoussaoui/oo7/pull/150#discussion_r1858496398). `locked_unchanged` is there to track/filter-out objects that require a prompt to perform lock/unlock. If the object is already locked/unlocked we don't need a prompt, but returning their object paths is required. So, that why I pushed it to `locked_changed`.

Reply:
Then the naming is confusing, maybe we should use some different naming like

Expected state(when the state won't change) and changed state for the other case.

Reply:
Yeah, make sense.

Reply:
is `locked` and `locked_changed` any good?

Reply:
not really because they depend on the value of the passed parameter